src/client/scripts/game/gui/guipractice.js

```diff
@@ -0,0 +1,213 @@
+ * our practice selection menu.
+ */
+
+"use strict";
+
+const guipractice = (function(){
+
+    // Variables
+
+    const element_menuExternalLinks = document.getElementById('menu-external-links');
+
+    const element_practiceSelection = document.getElementById('practice-selection')
+    const element_practiceName = document.getElementById('practice-name')
+    const element_practiceBack = document.getElementById('practice-back')
+    const element_checkmatePractice = document.getElementById('checkmate-practice')
+    const element_tacticsPractice = document.getElementById('tactics-practice')
+    const element_practicePlay = document.getElementById('practice-play')
+
+    const elements_checkmates = document.getElementsByClassName('checkmate');
+
+    let modeSelected; // checkmate-practice / tactics-practice
+    let checkmateSelectedID; // id of selected checkmate
+
+    // Functions
+
+    function getModeSelected() {
+        return modeSelected;
+    }
+
+    function open() {
+        gui.setScreen('title practice')
+        style.revealElement(element_practiceSelection)
+        style.revealElement(element_menuExternalLinks);
+        changePracticeMode('checkmate-practice')
+        changeCheckmateSelected('2Q-1k')
+        updateCheckmatesBeaten()
+        initListeners()
+    }
+
+    function close() {
+        style.hideElement(element_practiceSelection)
+        style.hideElement(element_menuExternalLinks);
+        closeListeners()
+    }
+
+    function initListeners() {
+        element_practiceBack.addEventListener('click', callback_practiceBack)
+        element_checkmatePractice.addEventListener('click', callback_checkmatePractice)
+        element_tacticsPractice.addEventListener('click', gui.callback_featurePlanned)
+        element_practicePlay.addEventListener('click', callback_practicePlay)
+        for (let element of elements_checkmates) {
+            element.addEventListener('click', callback_checkmateList);
+        }
+    }
+
+    function closeListeners() {
+        element_practiceBack.removeEventListener('click', callback_practiceBack)
+        element_checkmatePractice.removeEventListener('click', callback_checkmatePractice)
+        element_tacticsPractice.removeEventListener('click', gui.callback_featurePlanned)
+        element_practicePlay.removeEventListener('click', callback_practicePlay)
+        for (let element of elements_checkmates) {
+            element.removeEventListener('click', callback_checkmateList);
+        }
+    }
+
+    function changePracticeMode(mode) { // checkmate-practice / tactics-practice
+        modeSelected = mode
+        if (mode === 'checkmate-practice') {
+            element_practiceName.textContent = translations["menu_checkmate"]
+            element_checkmatePractice.classList.add('selected')
+            element_tacticsPractice.classList.remove('selected')
+            element_checkmatePractice.classList.remove('not-selected')
```

Comment:
Seems weird to have classes for both selected and not-selected, can we only use 1, selected, and assume elements without this class are not selected?

Reply:
Due to how specificity works in css, the selected class's selectors can overwrite the default ones